scripts/devops_tasks/common_tasks.py

```diff
@@ -492,3 +492,74 @@ def get_package_properties(setup_py_path):
     pkgName, version, _, requires = parse_setup(setup_py_path)
     is_new_sdk = pkgName in NEW_REQ_PACKAGES or any(map(lambda x: (parse_require(x)[0] in NEW_REQ_PACKAGES), requires))
     return pkgName, version, is_new_sdk, setup_py_path
+
+def is_track2_package(reqs):
+    for req in reqs:
+        # the package must be dependant on azure core or azure mgmt core
+        if req.startswith('azure-core') or req.startswith('azure-mgmt-core'):
+            return True
+    return False
+
+def get_track2_package_properties(setup_py_path):
+    """Parse setup.py and return package details like package name, version, whether it's new SDK
+    """
+    pkg_name, version, _, requires = parse_setup(setup_py_path)
+    is_track2 = is_track2_package(requires)
+    return pkg_name, version, is_track2, setup_py_path
+
+def get_all_track2_packages(path):
+    eligible_libraries = []
+    for root, dirs, files in os.walk(os.path.abspath(path)):
+        if re.search(r"sdk[\\/][^\\/]+[\\/][^\\/]+$", root):
+            if "setup.py" in files:
+                try:
+                    pkg_name, version, is_track2, setup_py_path = get_track2_package_properties(root)
+                    if is_track2:
+                        eligible_libraries.append((pkg_name, version, setup_py_path))
+                except:
+                    # Skip setup.py if the package cannot be parsed
+                    pass
+    return eligible_libraries
+
+def get_package_details(setup_filename):
```

Comment:
This already exists in `common_tasks.py`.

After my refactor, it will exist in azure-sdk-tools as well. But for now, this is duplicate code. If you need additional properties logged out of it, I highly recommend extending existing.